From 338b2da2d6937a9e39a67a5361ced01b88dc77ca Mon Sep 17 00:00:00 2001
From: Nisarg Jhaveri <nisargjhaveri@gmail.com>
Date: Tue, 16 May 2023 07:44:57 +0000
Subject: [PATCH] Fix move constructor for Variant

Move constuctor for Variant was removing the m_signature from both new and other object.
Use std::move for the m_signature member to get the correct behavior.

This was causing wrong message generated for method calls when the Variants were created and passed inline, causing the move constructor to be used.
---
 dbus-cxx/variant.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbus-cxx/variant.cpp b/dbus-cxx/variant.cpp
index 24581fd..58101e7 100644
--- a/dbus-cxx/variant.cpp
+++ b/dbus-cxx/variant.cpp
@@ -135,7 +135,7 @@ Variant::Variant( const Variant& other ) :
 
 Variant::Variant( Variant&& other ) :
     m_currentType( std::exchange( other.m_currentType, DataType::INVALID ) ),
-    m_signature( std::exchange( other.m_signature, "" ) ),
+    m_signature( std::move( other.m_signature ) ),
     m_marshaled( std::move( other.m_marshaled ) ),
     m_dataAlignment( std::exchange( other.m_dataAlignment, 0 ) ){
 }
-- 
2.34.1

